### PR TITLE
EPMDJ-10318 Added ability to store field in separate column

### PR DESCRIPTION
**Pull request by ViktoriiaLebedeva**

### CODE REVIEW

File: src/main/kotlin/util/DbUtil.kt [hunk 1]
```diff
@@ -166,3 +201,14 @@ fun SerialDescriptor.collectionPaths(path: String = ""): List<PathToTable> {
 fun List<PathToTable>.toQuery() = fold("") { acc, (path, table) ->
     acc + "DELETE FROM $table WHERE $ID_COLUMN IN (SELECT * FROM json_array_elements_text((OLD.json_body$path)::json)); \n"
 }
+
+fun Map<Column, List<String>>.fillColumn() = entries.fold("") { acc, (annotation, path) ->
+    acc + "NEW.${annotation.name} = CAST(NEW.json_body${path.build()} as ${annotation.type.value}); \n"
+}
+
+fun List<String>.build(
```
Comment (ViktoriiaLebedeva) [inline src/main/kotlin/util/DbUtil.kt:209]: toSqlPath

File: src/main/kotlin/find/Expr.kt [hunk 2]
```diff
@@ -17,29 +17,35 @@ package com.epam.dsm.find
...
     }
+}
+
+class ColumnPath(
+    vararg properties: KProperty1<*, *>
+) : Path(properties.map { it.findAnnotation<Column>()?.name ?: it.name }) {
 
-    constructor(fields: List<String>) {
-        path = fields
+    override fun buildSqlPath(lastOperation: String): String {
```
Comment (ViktoriiaLebedeva) [inline src/main/kotlin/find/Expr.kt:42]: javadoc

File: src/main/kotlin/util/DbUtil.kt [hunk 3]
```diff
@@ -135,6 +147,29 @@ fun String.createTrigger(
     """.trimIndent()
 }
 
+fun insertTrigger(tableName: String, columns: Map<Column, List<String>>): String = run {
```
Comment (ViktoriiaLebedeva) [inline src/main/kotlin/util/DbUtil.kt:150]: javadoc

File: src/main/kotlin/util/Util.kt [hunk 4]
```diff
@@ -51,6 +51,28 @@ fun idNames(desc: SerialDescriptor) = (0 until desc.elementsCount).filter { inde
...
+    val element = getElementName(index)
+    getElementAnnotations(index).find { it is Column }?.let {
+        result[it as Column] = path + element
+    }
+    if (innerDescriptor.isClassSerialKind()) {
+        innerDescriptor.findColumnAnnotation(path + element, result)
+    } else return@forEach
+}.run { result }
+
+fun SerialDescriptor.buildJson() = findColumnAnnotation().entries.fold(JSON_COLUMN) { acc, (annotation, path) ->
```
Comment (ViktoriiaLebedeva) [inline src/main/kotlin/util/Util.kt:68]: to DbUtil

File: src/main/kotlin/util/Util.kt [hunk 5]
```diff
@@ -51,6 +51,28 @@ fun idNames(desc: SerialDescriptor) = (0 until desc.elementsCount).filter { inde
...
+    """jsonb_set($acc, '{${path.joinToString()}}'::text[], to_jsonb(${annotation.name}), true)""".trimIndent()
+}
+
+inline fun <reified T : Any> fullJson(
```
Comment (ViktoriiaLebedeva) [inline src/main/kotlin/util/Util.kt:72]: move to DbUtil

File: src/main/kotlin/Core.kt [hunk 6]
```diff
@@ -126,9 +127,9 @@ suspend inline fun <reified T : Any> Transaction.getAll(): MutableList<T> {
     val finalData = mutableListOf<T>()
     val tableName = createTableIfNotExists<T>(connection.schema)
     val classLoader = T::class.java.classLoader
-    execWrapper("select $JSON_COLUMN FROM $tableName") { rs ->
+    execWrapper("select ${fullJson<T>()} FROM $tableName") { rs ->
         while (rs.next()) {
-            finalData.add(dsmDecode(rs.getBinaryStream(1), classLoader))
+            finalData.add(dsmDecode(rs.getString(1), classLoader))
```
Comment (LikhanovMaxim) [inline src/main/kotlin/Core.kt:132]: revert and check

File: src/test/kotlin/ColumnAnnotationTest.kt [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+    @Test
+    fun `simple class - should detect Column annotation`() {
+        val columnPaths = Simple::class.serializer().descriptor.findColumnAnnotation()
+        assertTrue { columnPaths.isNotEmpty() }
+        assertTrue { columnPaths.containsValue(listOf(Simple::separate.name)) }
+    }
+
+    @Test
+    fun `annotation in inner class - should detect path`() {
```
Comment (LikhanovMaxim) [inline src/test/kotlin/ColumnAnnotationTest.kt:37]: rename

**Pull request merged**